### PR TITLE
Add Linode Images builder

### PR DESCRIPTION
**Pull request by displague**
Packer Builder for [Linode Images](https://www.linode.com/docs/platform/disk-images/linode-images/)

Adds the following builder:

  * `linode`

Based on https://github.com/linode/packer-builder-linode (MPL/2)
(formerly maintained by @dradtke).  Includes website docs and tests.

Relates to #174, #3131

### CODE REVIEW

File: builder/linode/config.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	ImageLabel   string   `mapstructure:"image_label"`
+	Description  string   `mapstructure:"image_description"`
+
+	RawStateTimeout string `mapstructure:"state_timeout"`
+
+	stateTimeout time.Duration
+	interCtx     interpolate.Context
+}
+
+func createRandomRootPassword() (string, error) {
```
[the remark this answers is not in the document]
Comment (displague) [reply builder/linode/config.go]: Fair enough.  I use `crypto/rand` for stronger (temporary) passwords.  When images are deployed the `root_pass` argument used when creating the new instance (or disk) will replace this password in any case.

[the remark this answers is not in the document]
Comment (displague) [reply builder/linode/config.go]: I'll also add @ctreatma and @stvnjacobs, both of which are also at Linode, and have involvement in this project.

**Comment by scub** [comment]
WOOOOOOOOOOOOOOOO! :shipit:

**Comment by dg10a** [comment]
I'm running into an error at the last step of the build:
```
> packer build --var-file=secrets.json test.json
linode_node output will be in this color.

==> linode_node: Running builder ...
==> linode_node: Creating temporary SSH key for instance...
==> linode_node: Creating Linode...
==> linode_node: Using ssh communicator to connect: <IP>
==> linode_node: Waiting for SSH to become available...
==> linode_node: Connected to SSH!
==> linode_node: Shutting down Linode...
==> linode_node: Creating image...
==> linode_node: Error creating image: [404] Not found
Build 'linode_node' errored: Error creating image: [404] Not found

==> Some builds didn't complete successfully and had errors:
--> linode_node: Error creating image: [404] Not found

==> Builds finished but no artifacts were created.
```


Here's the build template:
```
{
  "builders": [
    {
      "name": "linode_node",
      "type": "linode",
      "linode_token": "{{ user `linode_token` }}",
      "image": "linode/arch",
      "region": "us-west",
      "ssh_username": "root",
      "instance_type": "g6-nanode-1"
    }
  ]
}
```

**Comment by displague** [comment]
@dg10a The 404 indicates that the image was not successfully captured.  This is generally caused by not having enough image size or count quota for the image being created.  

https://www.linode.com/docs/platform/disk-images/linode-images-classic-manager/#managing-your-images

Reach out to Linode support to have your quota increased.

**Pull request merged**